### DISCUSSION TITLE
Render Agent Report as a standalone file

### DISCUSSION THREAD

**Discussion by jkaupp** [rstudio/pointblank#284] [Q&A] [answered]
Thank you for your work in such a great package!  I've been searching for a way to have agent reports to be produced/saved as a standalone HTML file in a non-clunky fashion.

've been using:

```
sink("agent report.html")
print(out, view = FALSE)
sink()
```
Where `out` is my completed agent report.   There must be a better way to do this!  I would appreciate any guidance and was also wondering if this was an area for future development.

**Comment by rich-iannone** [comment, marked as answer]
Thanks for asking about this. I agree! It is not great currently and it would be better to get the reporting as an object you can integrate elsewhere (e.g., HTML fragment, HTML page, gt object, etc.). One thing you can do right now (and this is still not ideal) is something like this:

``` r
html_fragment <- agent %>% get_agent_report() %>% gt::as_raw_html(inline_css = FALSE)
```

But this doesn't give you a standalone HTML doc, but, you can use htmltools to put the HTML in an element. I'll follow up shortly with an example of that (once I've worked it out), and, I'll want to improve pointblank to improve `get_agent_report()` so you get the object you really want.

**Reply by rich-iannone** [reply]
As an update to this, I found some ways to improve the raw HTML output of {gt} tables in gt. This is important because these reports are in fact {gt} tables. We are working on a new release of {gt} and I hope to get these improvements for exporting in that release. Then, {pointblank} will directly benefit since I can improve `get_agent_report()` without resorting to hacks.

**Reply by rich-iannone** [reply]
Another update, one that I hope closes the loop on this: the new `export_report()` function (currently in the dev version of the package) makes it possible to easily write reports to HTML. The online docs for this are here: https://rich-iannone.github.io/pointblank/reference/export_report.html. This took awhile because improvements to gt were needed. However, this hopefully gives you what you need. Please let me know if it works for you and whether it can be improved!
